pkg/controller/infrastructure/actuator.go

```diff
@@ -507,6 +536,10 @@ func (a *actuator) reconcile(ctx context.Context, infra *extensionsv1alpha1.Infr
 		return err
 	}
 
+	if err = a.ensureOldSSHKeyDetached(ctx, infra); err != nil {
```

Comment:
Why this func is invoked exactly here? I guess the more appropriate place would be before creating the initializer.

```suggestion
	if err = a.ensureOldSSHKeyDetached(ctx, infra); err != nil {
		return err
	}

	tf, err := common.NewTerraformerWithAuth(a.logger, a.terraformerFactory, a.RESTConfig(), TerraformerPurpose, infra)
	if err != nil {
		return err
	}

	initializerValues, err := a.getInitializerValues(ctx, tf, infra, config, credentials)
	if err != nil {
		return err
	}

	initializer, err := a.newInitializer(infra, config, initializerValues, stateInitializer)
	if err != nil {
		return err
	}

	if err := tf.InitializeWith(ctx, initializer).Apply(ctx); err != nil {
		return errors.Wrapf(err, "failed to apply the terraform config")
	}
```

Reply:
I don't have any preference. I originally tried to ensureOldSSHKeyDetached before terraformer logic segment instead of terraformer execution. If you have no objection, I would keep it unchanged.

Reply:
I think the better approach is the one proposed by me. The tf init preparations should happen before the Apply operation. Currently there is an operation (`ensureOldSSHKeyDetached`) between the tf init preparations and the Apply itself.

Reply:
I will take your advise. I messed your proposal with my real implementation from this change proposal.  sorry~